contracts/button/lib.rs

```diff
@@ -261,10 +299,7 @@ pub mod button_game {
             &mut self,
             new_button_lifetime: BlockNumber,
         ) -> ButtonResult<()> {
-            let caller = self.env().caller();
-            let this = self.env().account_id();
-            let required_role = Role::Admin(this);
-            ButtonGame::check_role(&self.access_control, caller, required_role)?;
+            self.check_role(self.env().caller(), Role::Admin(self.env().account_id()))?;
```

Comment:
We're not checking for halted state here b/c it can only be called by an admin?

Reply:
yes, see previous responses